---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "README-"
)
```

# UCSCXenaTools <img src='man/figures/logo.png' align="right" height="140" width="120" alt="logo"/>

<!-- badges: start -->
[![CRAN](http://www.r-pkg.org/badges/version-last-release/UCSCXenaTools)](https://cran.r-project.org/package=UCSCXenaTools) [![AppVeyor Build Status](https://ci.appveyor.com/api/projects/status/github/ShixiangWang/UCSCXenaTools?branch=master&svg=true)](https://ci.appveyor.com/project/ShixiangWang/UCSCXenaTools) [![Travis build status](https://travis-ci.org/ShixiangWang/UCSCXenaTools.svg?branch=master)](https://travis-ci.org/ShixiangWang/UCSCXenaTools)

![](http://cranlogs.r-pkg.org/badges/UCSCXenaTools) ![](http://cranlogs.r-pkg.org/badges/grand-total/UCSCXenaTools) [![Coverage Status](https://img.shields.io/codecov/c/github/ShixiangWang/UCSCXenaTools/master.svg)](https://codecov.io/github/ShixiangWang/UCSCXenaTools?branch=master) [![GitHub issues](https://img.shields.io/github/issues/ShixiangWang/UCSCXenaTools.svg)](https://github.com/ShixiangWang/UCSCXenaTools/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+) [![Closed issues](https://img.shields.io/github/issues-closed/ShixiangWang/UCSCXenaTools.svg)](https://github.com/ShixiangWang/UCSCXenaTools/issues?q=is%3Aissue+is%3Aclosed)
<!-- badges: end -->

**UCSCXenaTools** is a R package download and explore data from **UCSC Xena data hubs**, which are

> a collection of UCSC-hosted public databases such as TCGA, ICGC, TARGET, GTEx, CCLE, and others. Databases are normalized so they can be combined, linked, filtered, explored and downloaded.
>
> -- [UCSC Xena](https://xena.ucsc.edu/)

If you use this package in academic field, please cite:

*Wang, Shixiang, et al. "APOBEC3B and APOBEC mutational signature as potential predictive markers for immunotherapy response in non-small cell lung cancer." Oncogene (2018).*

## Installation

Install stable release from CRAN with:

```{r, eval=FALSE}
install.packages("UCSCXenaTools")
```


You can also install devel version of **UCSCXenaTools** from github with:

```{r gh-installation, eval = FALSE}
# install.packages("remotes")
remotes::install_github("ShixiangWang/UCSCXenaTools", build_vignettes = TRUE)
```

Read this vignettes.

```{r, eval=FALSE}
browseVignettes("UCSCXenaTools")
# or
??UCSCXenaTools
```


## Data Hub List

All datasets are available at <https://xenabrowser.net/datapages/>.

Currently, **UCSCXenaTools** support all 9 data hubs of UCSC Xena.

* UCSC Public Hub: <https://ucscpublic.xenahubs.net>
* TCGA Hub: <https://tcga.xenahubs.net>
* GDC Xena Hub: <https://gdc.xenahubs.net>
* ICGC Xena Hub: <https://icgc.xenahubs.net>
* Pan-Cancer Atlas Hub: <https://pancanatlas.xenahubs.net>
* GA4GH (TOIL) Hub: <https://toil.xenahubs.net>
* Treehouse Hub: <https://xena.treehouse.gi.ucsc.edu>
* PCAWG: <https://pcawg.xenahubs.net>
* ATAC-seq: <https://atacseq.xenahubs.net>

If the url of data hub changed, please remind me by emailing to <w_shixiang@163.com> or [opening an issue on GitHub](https://github.com/ShixiangWang/UCSCXenaTools/issues).

## Usage

Download UCSC Xena Datasets and load them into R by **UCSCXenaTools** is a workflow in `generate`, `filter`, `query`, `download` and `prepare` 5 steps, which are implemented as `XenaGenerate`, `XenaFilter`, `XenaQuery`, `XenaDownload` and `XenaPrepare` functions, respectively. They are very clear and easy to use and combine with other packages like `dplyr`.

To show the basic usage of **UCSCXenaTools**, we download clinical data of LUNG, LUAD, LUSC from TCGA (hg19 version) data hub.

### XenaData data.frame

Begin from version `0.2.0`, **UCSCXenaTools** uses a `data.frame` object (built in package) `XenaData` to generate an instance of `XenaHub` class, which communicate with API of UCSC Xena Data Hubs.

You can load `XenaData` after loading `UCSCXenaTools` into R.

```{r}
library(UCSCXenaTools)
data(XenaData)

head(XenaData)
```

### Generate a XenaHub object

This can be implemented by `XenaGenerate` function, which generate `XenaHub` object from `XenaData` data frame.

```{r}
XenaGenerate()
```

You can set `subset` argument to narrow datasets.

```{r}
XenaGenerate(subset = XenaHostNames=="TCGA")
```

>You can also use `XenaHub()` to generate a `XenaHub` object for API communication, but it is not recommended. 

It's possible to explore `hosts()`, `cohorts()` and `datasets()`.

```{r}
xe = XenaGenerate(subset = XenaHostNames=="TCGA")
# get hosts
hosts(xe)
# get cohorts
head(cohorts(xe))
# get datasets
head(datasets(xe))
```

Pipe operator `%>%` can also be used here.

```{r, message=FALSE}
library(dplyr)
XenaData %>% 
    filter(XenaHostNames == "TCGA", grepl("BRCA", XenaCohorts), grepl("Path", XenaDatasets)) %>%
    XenaGenerate()
```

### Filter 

There are too many datasets in `xe`, you can filter them by `XenaFilter` function.Regular expression can be used here.

```{r}
(XenaFilter(xe, filterDatasets = "clinical") -> xe2)
```


Then select `LUAD`, `LUSC` and `LUNG` 3 datasets.

```{r}
XenaFilter(xe2, filterDatasets = "LUAD|LUSC|LUNG") -> xe2
```

Pipe can be used here.

```{r}
xe %>% 
    XenaFilter(filterDatasets = "clinical") %>% 
    XenaFilter(filterDatasets = "luad|lusc|lung")
```

### Query

Create a query before downloading data.

```{r}
xe2_query = XenaQuery(xe2)
xe2_query
```

### Download

Default, data will be downloaded to system temp directory. You can specify the path.

If the data exists, command will not run to download them, but you can force it by `force` option.

```{r}
xe2_download = XenaDownload(xe2_query)
```

> **Note** file names inherit from names in datasets column and '/' all changed to '__'.

### Prepare

There are 4 ways to prepare data to R.

```
# way1:  directory
cli1 = XenaPrepare("E:/Github/XenaData/test/")
names(cli1)
## [1] "TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz"
## [2] "TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz"
## [3] "TCGA.LUSC.sampleMap__LUSC_clinicalMatrix.gz"
```

```
# way2: local files
cli2 = XenaPrepare("E:/Github/XenaData/test/TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz")
class(cli2)
## [1] "tbl_df"     "tbl"        "data.frame"

cli2 = XenaPrepare(c("E:/Github/XenaData/test/TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz",
                     "E:/Github/XenaData/test/TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz"))
class(cli2)
## [1] "list"
names(cli2)
## [1] "TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz"
## [2] "TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz"
```

```
# way3: urls
cli3 = XenaPrepare(xe2_download$url[1:2])
names(cli3)
## [1] "LUSC_clinicalMatrix.gz" "LUNG_clinicalMatrix.gz"
```

```{r}
# way4: xenadownload object
cli4 = XenaPrepare(xe2_download)
names(cli4)
```

From v0.2.6, `XenaPrepare()` can enable chunk feature when file is too big and user only need subset of file.

Following code show how to subset some rows or columns of files, `sample` is the name of the first column, user can directly use it in logical expression,  `x` can be a representation of data frame user wanna do subset operation. More custom operation can be set as a function and pass to `callback` option.

```
# select rows which sample (gene symbol here) in "HIF3A" or "RNF17"
testRNA = UCSCXenaTools::XenaPrepare("~/Download/HiSeqV2.gz", use_chunk = TRUE, subset_rows = sample %in% c("HIF3A", "RNF17"))
# only keep 1 to 3 columns
testRNA = UCSCXenaTools::XenaPrepare("~/Download/HiSeqV2.gz", use_chunk = TRUE, select_cols = colnames(x)[1:3])
```

## TCGA Common Data Easy Download

### getTCGAdata

`getTCGAdata` provide a quite easy download way for TCGA datasets, user can specify multiple options to select which data and corresponding file type want to download. Default this function will return a list include `XenaHub` object and selective datasets information. Once you are sure the datasets is exactly what you want, `download` can be set to `TRUE` to download the data.

Check arguments of `getTCGAdata`:

```{r}
args(getTCGAdata)

# or run
# ??getTCGAdata to read documentation
```

Select one or more projects, default will select only clinical datasets:

```{r}
getTCGAdata(c("UVM", "LUAD"))

tcga_data = getTCGAdata(c("UVM", "LUAD"))

# only return XenaHub object
tcga_data$Xena

# only return datasets information
tcga_data$DataInfo
```

Set `download=TRUE` to download data, default data will be downloaded to system temp directory (you can specify the path with `destdir` option):

```{r, eval=FALSE}
# only download clinical data
getTCGAdata(c("UVM", "LUAD"), download = TRUE)
```

#### Support Data Type and Options

* clinical information: `clinical`
* mRNA Sequencing: `mRNASeq`
* mRNA microarray: `mRNAArray`
* miRNA Sequencing: `miRNASeq`
* exon Sequencing: `exonRNASeq`
* RPPA array: `RPPAArray`
* DNA Methylation: `Methylation`
* Gene mutation: `GeneMutation`
* Somatic mutation: `SomaticMutation`
* Gistic2 Copy Number: `GisticCopyNumber`
* Copy Number Segment: `CopyNumberSegment`

> other data type supported by Xena cannot download use this function. Please refer to `downloadTCGA` function or `XenaGenerate` function.


**NOTE**: Sequencing data are all based on  Illumina Hiseq platform, other platform (Illumina GA) data supported by Xena cannot download using this function. This is for building consistent data download flow. Mutation use `broad automated` version (except `PANCAN` use `MC3 Public Version`). If you wan to download other datasets, please refer to `downloadTCGA` function or `XenaGenerate` function.


### download any TCGA data by datatypes and filetypes 

`downloadTCGA` function can used to download any TCGA data supported by Xena, but in a way different from `getTCGAdata` function.

```{r, eval=FALSE}
# download RNASeq data (use UVM as an example)
downloadTCGA(project = "UVM",
                 data_type = "Gene Expression RNASeq",
                  file_type = "IlluminaHiSeq RNASeqV2")
```


See the arguments:

```{r}
args(downloadTCGA)
```

Except `destdir` option, you only need to select three arguments for downloading data. Even throught the number is far less than `getTCGAdata`, it is more complex than the latter.

Before you download data, you need spare some time to figure out what data type and file type available and what your datasets have.

`availTCGA` can return all information you need:

```{r}
availTCGA()
```

Note not all datasets have these property, `showTCGA` can help you to check it. It will return all data in TCGA, you can use following code in RStudio and search your data.

```{r, eval=FALSE}
View(showTCGA())
```

**OR you can use shiny app provided by `UCSCXenaTools` to search**.

Run shiny by:

```{r, eval=FALSE}
UCSCXenaTools::XenaShiny()
```

Download by shiny is under consideration, I am try learning more about how to operate shiny.


## SessionInfo

```{r}
sessionInfo()
```

## Bug Report

I have no time to test if all condition are right and all datasets can normally be downloaded. So if you have any question or suggestion, please open an issue on Github at <https://github.com/ShixiangWang/UCSCXenaTools/issues>. 

## Acknowledgement

This package is based on [XenaR](https://github.com/mtmorgan/XenaR), thanks [Martin Morgan](https://github.com/mtmorgan) for his work.

## LICENSE

GPL-3

please note, code from XenaR package under Apache 2.0 license.

## Code of conduct

Please note that this project is released with a [Contributor Code of
Conduct](CODE_OF_CONDUCT.md). By participating in this project you agree
to abide by its terms.
