<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Download and Explore Datasets from UCSC Xena Data Hubs • UCSCXenaTools</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Download and Explore Datasets from UCSC Xena Data Hubs">
<meta property="og:description" content="Download and explore datasets from UCSC Xena data hubs, which are
    a collection of UCSC-hosted public databases such as TCGA, ICGC, TARGET, GTEx, CCLE, and others.
    Databases are normalized so they can be combined, linked, filtered, explored and downloaded.">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">UCSCXenaTools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.4.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/USCSXenaTools.html">UCSCXenaTools: an R package for Accessing Genomics Data from UCSC Xena platform, from Cancer Multi-omics to Single-cell RNA-seq</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ropensci/UCSCXenaTools/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<table class="table table">
<thead><tr class="header">
<th align="left">
Usage
</th>
<th align="left">
Release
</th>
<th align="left">
Development
</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">
<a href="https://www.gnu.org/licenses/gpl-3.0.en.html"><img src="https://img.shields.io/badge/licence-GPL--3-blue.svg" alt="Licence"></a>
</td>
<td align="left">
<a href="https://cran.r-project.org/package=UCSCXenaTools"><img src="https://www.r-pkg.org/badges/version/UCSCXenaTools" alt="CRAN"></a>
</td>
<td align="left">
<a href="https://travis-ci.org/ropensci/UCSCXenaTools"><img src="https://travis-ci.org/ropensci/UCSCXenaTools.svg?branch=master" alt="Travis"></a>
</td>
</tr>
<tr class="even">
<td align="left">
<a href="https://cran.r-project.org/"><img src="https://img.shields.io/badge/R%3E%3D-3.5.0-blue.svg" alt="minimal R version"></a>
</td>
<td align="left">
<a href="https://cran.r-project.org/web/checks/check_results_UCSCXenaTools.html"><img src="https://cranchecks.info/badges/summary/UCSCXenaTools" alt="cran-checks"></a>
</td>
<td align="left">
<a href="https://ci.appveyor.com/project/ShixiangWang/UCSCXenaTools"><img src="https://ci.appveyor.com/api/projects/status/github/ropensci/UCSCXenaTools?branch=master&amp;svg=true" alt="AppVeyor"></a>
</td>
</tr>
<tr class="odd">
<td align="left">
<a href="https://CRAN.R-project.org/package=UCSCXenaTools"><img src="https://tinyverse.netlify.com/badge/UCSCXenaTools"></a>
</td>
<td align="left">
<a href="https://github.com/ropensci/software-review/issues/315"><img src="https://badges.ropensci.org/315_status.svg" alt="rOpenSci"></a>
</td>
<td align="left">
<a href="https://codecov.io/github/ShixiangWang/UCSCXenaTools?branch=master"><img src="https://codecov.io/github/ShixiangWang/UCSCXenaTools/coverage.svg?branch=master" alt="Codecov"></a>
</td>
</tr>
<tr class="even">
<td align="left">
<a href="https://CRAN.R-project.org/package=UCSCXenaTools"><img src="https://cranlogs.r-pkg.org/badges/grand-total/UCSCXenaTools" alt="downloads"></a>
</td>
<td align="left">
<a href="https://zenodo.org/badge/latestdoi/178662770"><img src="https://zenodo.org/badge/178662770.svg" alt="DOI"></a>
</td>
<td align="left">
<a href="https://github.com/ropensci/UCSCXenaTools/issues?q=is%3Aissue+is%3Aclosed"><img src="https://img.shields.io/github/issues-closed/ropensci/UCSCXenaTools.svg" alt="Closed issues"></a>
</td>
</tr>
<tr class="odd">
<td align="left">
<a href="https://CRAN.R-project.org/package=UCSCXenaTools"><img src="https://cranlogs.r-pkg.org/badges/UCSCXenaTools" alt="month-downloads"></a>
</td>
<td align="left">
<a href="https://doi.org/10.21105/joss.01627"><img src="https://joss.theoj.org/papers/10.21105/joss.01627/status.svg" alt="JOSS"></a>
</td>
<td align="left">
<a href="https://www.repostatus.org/#active"><img src="https://www.repostatus.org/badges/latest/active.svg" alt="Project Status: Active – The project has reached a stable, usable state and is being actively developed."></a>
</td>
</tr>
</tbody>
</table>
<p><br><!-- badges: end --></p>
<div id="ucscxenatools-" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#ucscxenatools-" class="anchor"></a>UCSCXenaTools <img src="reference/figures/logo.png" align="right" height="200" alt="logo">
</h1></div>
<p><strong>UCSCXenaTools</strong> is an R package for accessing genomics data from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq. Public omics data from UCSC Xena are supported through <a href="https://xenabrowser.net/datapages/"><strong>multiple turn-key Xena Hubs</strong></a>, which are a collection of UCSC-hosted public databases such as TCGA, ICGC, TARGET, GTEx, CCLE, and others. Databases are normalized so they can be combined, linked, filtered, explored and downloaded.</p>
<p><strong>Who is the target audience and what are scientific applications of this package?</strong></p>
<ul>
<li>Target Audience: cancer and clinical researchers, bioinformaticians</li>
<li>Applications: genomic and clinical analyses</li>
</ul>
<div id="table-of-contents" class="section level2">
<h2 class="hasAnchor">
<a href="#table-of-contents" class="anchor"></a>Table of Contents</h2>
<ul>
<li><a href="#installation">Installation</a></li>
<li><a href="#data-hub-list">Data Hub List</a></li>
<li><a href="#basic-usage">Basic usage</a></li>
<li><a href="#citation">Citation</a></li>
<li><a href="#how-to-contribute">How to contribute</a></li>
<li><a href="#acknowledgment">Acknowledgment</a></li>
</ul>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>Install stable release from CRAN with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"UCSCXenaTools"</span><span class="op">)</span></code></pre></div>
<p>You can also install devel version of <strong>UCSCXenaTools</strong> from github with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("remotes")</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"ropensci/UCSCXenaTools"</span><span class="op">)</span></code></pre></div>
<p>If you want to build vignette in local, please add two options:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"ropensci/UCSCXenaTools"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div id="data-hub-list" class="section level2">
<h2 class="hasAnchor">
<a href="#data-hub-list" class="anchor"></a>Data Hub List</h2>
<p>All datasets are available at <a href="https://xenabrowser.net/datapages/" class="uri">https://xenabrowser.net/datapages/</a>.</p>
<p>Currently, <strong>UCSCXenaTools</strong> supports the following data hubs of UCSC Xena.</p>
<ul>
<li>UCSC Public Hub: <a href="https://ucscpublic.xenahubs.net/" class="uri">https://ucscpublic.xenahubs.net/</a>
</li>
<li>TCGA Hub: <a href="https://tcga.xenahubs.net/" class="uri">https://tcga.xenahubs.net/</a>
</li>
<li>GDC Xena Hub: <a href="https://gdc.xenahubs.net/" class="uri">https://gdc.xenahubs.net/</a>
</li>
<li>ICGC Xena Hub: <a href="https://icgc.xenahubs.net/" class="uri">https://icgc.xenahubs.net/</a>
</li>
<li>Pan-Cancer Atlas Hub: <a href="https://pancanatlas.xenahubs.net/" class="uri">https://pancanatlas.xenahubs.net/</a>
</li>
<li>UCSC Toil RNAseq Recompute Compendium Hub: <a href="https://toil.xenahubs.net/" class="uri">https://toil.xenahubs.net/</a>
</li>
<li>PCAWG Xena Hub: <a href="https://pcawg.xenahubs.net/" class="uri">https://pcawg.xenahubs.net/</a>
</li>
<li>ATAC-seq Hub: <a href="https://atacseq.xenahubs.net/" class="uri">https://atacseq.xenahubs.net/</a>
</li>
<li>Singel Cell Xena Hub: <a href="https://singlecellnew.xenahubs.net/" class="uri">https://singlecellnew.xenahubs.net/</a>
</li>
<li>Kids First Xena Hub: <a href="https://kidsfirst.xenahubs.net/" class="uri">https://kidsfirst.xenahubs.net/</a>
</li>
<li>Treehouse Xena Hub: <a href="https://xena.treehouse.gi.ucsc.edu:443/" class="uri">https://xena.treehouse.gi.ucsc.edu:443/</a>
</li>
</ul>
<p>Users can update dataset list from the newest version of UCSC Xena by hand with <code><a href="reference/XenaDataUpdate.html">XenaDataUpdate()</a></code> function, followed by restarting R and <code><a href="https://docs.ropensci.org/UCSCXenaTools/">library(UCSCXenaTools)</a></code>.</p>
<p>If any url of data hub is changed or a new data hub is online, please remind me by emailing to <a href="mailto:w_shixiang@163.com" class="email">w_shixiang@163.com</a> or <a href="https://github.com/ropensci/UCSCXenaTools/issues">opening an issue on GitHub</a>.</p>
</div>
<div id="basic-usage" class="section level2">
<h2 class="hasAnchor">
<a href="#basic-usage" class="anchor"></a>Basic usage</h2>
<p>Download UCSC Xena datasets and load them into R by <strong>UCSCXenaTools</strong> is a workflow with <code>generate</code>, <code>filter</code>, <code>query</code>, <code>download</code> and <code>prepare</code> 5 steps, which are implemented as <code>XenaGenerate</code>, <code>XenaFilter</code>, <code>XenaQuery</code>, <code>XenaDownload</code> and <code>XenaPrepare</code> functions, respectively. They are very clear and easy to use and combine with other packages like <code>dplyr</code>.</p>
<p>To show the basic usage of <strong>UCSCXenaTools</strong>, we will download clinical data of LUNG, LUAD, LUSC from TCGA (hg19 version) data hub. Users can learn more about <strong>UCSCXenaTools</strong> by running <code><a href="https://rdrr.io/r/utils/browseVignettes.html">browseVignettes("UCSCXenaTools")</a></code> to read vignette.</p>
<div id="xenadata-dataframe" class="section level3">
<h3 class="hasAnchor">
<a href="#xenadata-dataframe" class="anchor"></a>XenaData data.frame</h3>
<p><strong>UCSCXenaTools</strong> uses a <code>data.frame</code> object (built in package) <code>XenaData</code> to generate an instance of <code>XenaHub</code> class, which records information of all datasets of UCSC Xena Data Hubs.</p>
<p>You can load <code>XenaData</code> after loading <code>UCSCXenaTools</code> into R.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/UCSCXenaTools/">UCSCXenaTools</a></span><span class="op">)</span>
<span class="co">#&gt; =========================================================================================</span>
<span class="co">#&gt; UCSCXenaTools version 1.4.2</span>
<span class="co">#&gt; Project URL: https://github.com/ropensci/UCSCXenaTools</span>
<span class="co">#&gt; Usages: https://cran.r-project.org/web/packages/UCSCXenaTools/vignettes/USCSXenaTools.html</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; If you use it in published research, please cite:</span>
<span class="co">#&gt; Wang et al., (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data</span>
<span class="co">#&gt;   from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq.</span>
<span class="co">#&gt;   Journal of Open Source Software, 4(40), 1627, https://doi.org/10.21105/joss.01627</span>
<span class="co">#&gt; =========================================================================================</span>
<span class="co">#&gt;                               --Enjoy it--</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">XenaData</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">XenaData</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 17</span>
<span class="co">#&gt;   XenaHosts XenaHostNames XenaCohorts XenaDatasets SampleCount DataSubtype Label</span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;              &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;</span>
<span class="co">#&gt; 1 https://… publicHub     Breast Can… ucsfNeve_pu…          51 gene expre… Neve…</span>
<span class="co">#&gt; 2 https://… publicHub     Breast Can… ucsfNeve_pu…          57 phenotype   Phen…</span>
<span class="co">#&gt; 3 https://… publicHub     Glioma (Ko… kotliarov20…         194 copy number Kotl…</span>
<span class="co">#&gt; 4 https://… publicHub     Glioma (Ko… kotliarov20…         194 phenotype   Phen…</span>
<span class="co">#&gt; 5 https://… publicHub     Lung Cance… weir2007_pu…         383 copy number CGH  </span>
<span class="co">#&gt; 6 https://… publicHub     Lung Cance… weir2007_pu…         383 phenotype   Phen…</span>
<span class="co">#&gt; # … with 10 more variables: Type &lt;chr&gt;, AnatomicalOrigin &lt;chr&gt;,</span>
<span class="co">#&gt; #   SampleType &lt;chr&gt;, Tags &lt;chr&gt;, ProbeMap &lt;chr&gt;, LongTitle &lt;chr&gt;,</span>
<span class="co">#&gt; #   Citation &lt;chr&gt;, Version &lt;chr&gt;, Unit &lt;chr&gt;, Platform &lt;chr&gt;</span></code></pre></div>
</div>
<div id="workflow" class="section level3">
<h3 class="hasAnchor">
<a href="#workflow" class="anchor"></a>Workflow</h3>
<p>Select datasets.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The options in XenaFilter function support Regular Expression</span>
<span class="fu"><a href="reference/XenaGenerate.html">XenaGenerate</a></span><span class="op">(</span>subset <span class="op">=</span> <span class="va">XenaHostNames</span><span class="op">==</span><span class="st">"tcgaHub"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="reference/XenaFilter.html">XenaFilter</a></span><span class="op">(</span>filterDatasets <span class="op">=</span> <span class="st">"clinical"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="reference/XenaFilter.html">XenaFilter</a></span><span class="op">(</span>filterDatasets <span class="op">=</span> <span class="st">"LUAD|LUSC|LUNG"</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">df_todo</span>

<span class="va">df_todo</span>
<span class="co">#&gt; class: XenaHub </span>
<span class="co">#&gt; hosts():</span>
<span class="co">#&gt;   https://tcga.xenahubs.net</span>
<span class="co">#&gt; cohorts() (3 total):</span>
<span class="co">#&gt;   TCGA Lung Cancer (LUNG)</span>
<span class="co">#&gt;   TCGA Lung Adenocarcinoma (LUAD)</span>
<span class="co">#&gt;   TCGA Lung Squamous Cell Carcinoma (LUSC)</span>
<span class="co">#&gt; datasets() (3 total):</span>
<span class="co">#&gt;   TCGA.LUNG.sampleMap/LUNG_clinicalMatrix</span>
<span class="co">#&gt;   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</span>
<span class="co">#&gt;   TCGA.LUSC.sampleMap/LUSC_clinicalMatrix</span></code></pre></div>
<p>Query and download.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/XenaQuery.html">XenaQuery</a></span><span class="op">(</span><span class="va">df_todo</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="reference/XenaDownload.html">XenaDownload</a></span><span class="op">(</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">xe_download</span></code></pre></div>
<p><strong>For researchers in China, now <a href="https://hiplot.com.cn/">Hiplot team</a> has deployed several Xena <a href="https://xena.hiplot.com.cn/">mirror sites</a> at Shanghai. You can set an option <code><a href="https://rdrr.io/r/base/options.html">options(use_hiplot = TRUE)</a></code> before querying data step to speed up both data querying and downloading.</strong></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>use_hiplot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="fu"><a href="reference/XenaQuery.html">XenaQuery</a></span><span class="op">(</span><span class="va">df_todo</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="reference/XenaDownload.html">XenaDownload</a></span><span class="op">(</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">xe_download</span>
<span class="co">#&gt; Use hiplot server (China) for mirrored data hubs (set 'options(use_hiplot = FALSE)' to disable it)</span>
<span class="co">#&gt; This will check url status, please be patient.</span>
<span class="co">#&gt; All downloaded files will under directory /var/folders/mx/rfkl27z90c96wbmn3_kjk8c80000gn/T//Rtmp4UYCMN.</span>
<span class="co">#&gt; The 'trans_slash' option is FALSE, keep same directory structure as Xena.</span>
<span class="co">#&gt; Creating directories for datasets...</span>
<span class="co">#&gt; Downloading TCGA.LUNG.sampleMap/LUNG_clinicalMatrix</span>
<span class="co">#&gt; Downloading TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</span>
<span class="co">#&gt; Downloading TCGA.LUSC.sampleMap/LUSC_clinicalMatrix</span></code></pre></div>
<p>Prepare data into R for analysis.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cli</span> <span class="op">=</span> <span class="fu"><a href="reference/XenaPrepare.html">XenaPrepare</a></span><span class="op">(</span><span class="va">xe_download</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">cli</span><span class="op">)</span>
<span class="co">#&gt; [1] "list"</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">cli</span><span class="op">)</span>
<span class="co">#&gt; [1] "LUNG_clinicalMatrix" "LUAD_clinicalMatrix" "LUSC_clinicalMatrix"</span></code></pre></div>
</div>
</div>
<div id="more-to-read" class="section level2">
<h2 class="hasAnchor">
<a href="#more-to-read" class="anchor"></a>More to read</h2>
<ul>
<li><a href="https://shixiangwang.github.io/home/en/tools/ucscxenatools-intro/">Introduction and basic usage of UCSCXenaTools</a></li>
<li><a href="https://shixiangwang.github.io/home/en/post/ucscxenatools-201908/">UCSCXenaTools: Retrieve Gene Expression and Clinical Information from UCSC Xena for Survival Analysis</a></li>
<li><a href="https://shixiangwang.github.io/home/en/post/2020-07-22-ucscxenatools-single-gene/">Obtain RNAseq Values for a Specific Gene in Xena Database</a></li>
<li><a href="https://shixiangwang.github.io/home/en/tools/ucscxenatools-api/">UCSC Xena Access APIs in UCSCXenaTools</a></li>
</ul>
</div>
<div id="citation" class="section level2">
<h2 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h2>
<p>Cite me by the following paper.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>Wang et al., (<span class="dv">2019</span>). The UCSCXenaTools R package<span class="sc">:</span> a toolkit <span class="cf">for</span> accessing genomics data</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  from UCSC Xena platform, from cancer multi<span class="sc">-</span>omics to single<span class="sc">-</span>cell RNA<span class="sc">-</span>seq. </span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  Journal of Open Source Software, <span class="dv">4</span>(<span class="dv">40</span>), <span class="dv">1627</span>, https<span class="sc">:</span><span class="er">//</span>doi.org<span class="sc">/</span><span class="fl">10.21105</span><span class="sc">/</span>joss<span class="fl">.01627</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co"># For BibTex</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="sc">@</span>article{Wang2019UCSCXenaTools,</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>    journal <span class="ot">=</span> {Journal of Open Source Software},</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>    doi <span class="ot">=</span> {<span class="fl">10.21105</span><span class="sc">/</span>joss<span class="fl">.01627</span>},</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>    issn <span class="ot">=</span> {<span class="dv">2475-9066</span>},</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>    number <span class="ot">=</span> {<span class="dv">40</span>},</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>    publisher <span class="ot">=</span> {The Open Journal},</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>    title <span class="ot">=</span> {The UCSCXenaTools R package<span class="sc">:</span> a toolkit <span class="cf">for</span> accessing genomics data from UCSC Xena platform, from cancer multi<span class="sc">-</span>omics to single<span class="sc">-</span>cell RNA<span class="sc">-</span>seq},</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>    url <span class="ot">=</span> {https<span class="sc">:</span><span class="er">//</span>dx.doi.org<span class="sc">/</span><span class="fl">10.21105</span><span class="sc">/</span>joss<span class="fl">.01627</span>},</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>    volume <span class="ot">=</span> {<span class="dv">4</span>},</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>    author <span class="ot">=</span> {Wang, Shixiang and Liu, Xuesong},</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>    pages <span class="ot">=</span> {<span class="dv">1627</span>},</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>    date <span class="ot">=</span> {<span class="dv">2019-08-05</span>},</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>    year <span class="ot">=</span> {<span class="dv">2019</span>},</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>    month <span class="ot">=</span> {<span class="dv">8</span>},</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>    day <span class="ot">=</span> {<span class="dv">5</span>},</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>Cite UCSC Xena by the following paper.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>Goldman, Mary, et al. <span class="st">"The UCSC Xena Platform for cancer genomics data </span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="st">    visualization and interpretation."</span> <span class="fu">BioRxiv</span> (<span class="dv">2019</span>)<span class="sc">:</span> <span class="fl">326470.</span></span></code></pre></div>
</div>
<div id="how-to-contribute" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-contribute" class="anchor"></a>How to contribute</h2>
<p>For anyone who wants to contribute, please follow the guideline:</p>
<ul>
<li>Clone project from GitHub</li>
<li>Open <code>UCSCXenaTools.Rproj</code> with RStudio</li>
<li>Modify source code</li>
<li>Run <code><a href="https://devtools.r-lib.org//reference/check.html">devtools::check()</a></code>, and fix all errors, warnings and notes</li>
<li>Create a pull request</li>
</ul>
</div>
<div id="acknowledgment" class="section level2">
<h2 class="hasAnchor">
<a href="#acknowledgment" class="anchor"></a>Acknowledgment</h2>
<p>This package is based on <a href="https://github.com/mtmorgan/XenaR">XenaR</a>, thanks <a href="https://github.com/mtmorgan">Martin Morgan</a> for his work.</p>
<p><a href="https://ropensci.org"><img src="https://ropensci.org/public_images/ropensci_footer.png" alt="ropensci_footer"></a></p>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from CRAN at <br><a href="https://cloud.r-project.org/package=UCSCXenaTools">https://​cloud.r-project.org/​package=UCSCXenaTools</a>
</li>
<li>Browse source code at <br><a href="https://github.com/ropensci/UCSCXenaTools/">https://​github.com/​ropensci/​UCSCXenaTools/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/ropensci/UCSCXenaTools/issues">https://​github.com/​ropensci/​UCSCXenaTools/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></small></li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing UCSCXenaTools</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://shixiangwang.github.io/">Shixiang Wang</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-9855-7357" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Xue-Song Liu <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-7736-0077" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://shixiangwang.github.io/">Shixiang Wang</a>, Xue-Song Liu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
