<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>UCSCXenaTools: an R package for Accessing Genomics Data from UCSC Xena platform, from Cancer Multi-omics to Single-cell RNA-seq • UCSCXenaTools</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="UCSCXenaTools: an R package for Accessing Genomics Data from UCSC Xena platform, from Cancer Multi-omics to Single-cell RNA-seq">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">UCSCXenaTools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/USCSXenaTools.html">UCSCXenaTools: an R package for Accessing Genomics Data from UCSC Xena platform, from Cancer Multi-omics to Single-cell RNA-seq</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ropensci/UCSCXenaTools">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>UCSCXenaTools: an R package for Accessing Genomics Data from UCSC Xena platform, from Cancer Multi-omics to Single-cell RNA-seq</h1>
                        <h4 class="author">Shixiang Wang<br>
ShanghaiTech. University</h4>
            
            <h4 class="date">2020-02-26</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/UCSCXenaTools/blob/master/vignettes/USCSXenaTools.Rmd"><code>vignettes/USCSXenaTools.Rmd</code></a></small>
      <div class="hidden name"><code>USCSXenaTools.Rmd</code></div>

    </div>

    
    
<p><strong>UCSCXenaTools</strong> is an R package for accessing genomics data from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq. Public omics data from UCSC Xena are supported through <a href="https://xenabrowser.net/datapages/"><strong>multiple turn-key Xena Hubs</strong></a>, which are a collection of UCSC-hosted public databases such as TCGA, ICGC, TARGET, GTEx, CCLE, and others. Databases are normalized so they can be combined, linked, filtered, explored and downloaded.</p>
<p><strong>Who is the target audience and what are scientific applications of this package?</strong></p>
<ul>
<li>Target Audience: cancer and clinical researchers, bioinformaticians</li>
<li>Applications: genomic and clinical analyses</li>
</ul>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>Install stable release from CRAN with:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"UCSCXenaTools"</span>)</span></code></pre></div>
<p>You can also install devel version of <strong>UCSCXenaTools</strong> from github with:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="co"># install.packages("remotes")</span></span>
<span id="cb2-2"><a href="#cb2-2"></a>remotes<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/remotes/man/install_github.html">install_github</a></span>(<span class="st">"ropensci/UCSCXenaTools"</span>)</span></code></pre></div>
<p>If you want to build vignette in local, please add two options:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>remotes<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/remotes/man/install_github.html">install_github</a></span>(<span class="st">"ropensci/UCSCXenaTools"</span>, <span class="dt">build_vignettes =</span> <span class="ot">TRUE</span>, <span class="dt">dependencies =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p>The minimum versions to run the vignette is <code>1.2.4</code>. <a href="https://github.com/ropensci/UCSCXenaTools/issues">GitHub Issue</a> is a place for discussing any problem.</p>
</div>
<div id="data-hub-list" class="section level2">
<h2 class="hasAnchor">
<a href="#data-hub-list" class="anchor"></a>Data Hub List</h2>
<p>All datasets are available at <a href="https://xenabrowser.net/datapages/" class="uri">https://xenabrowser.net/datapages/</a>.</p>
<p>Currently, <strong>UCSCXenaTools</strong> supports the following data hubs of UCSC Xena.</p>
<ul>
<li>UCSC Public Hub: <a href="https://ucscpublic.xenahubs.net" class="uri">https://ucscpublic.xenahubs.net</a>
</li>
<li>TCGA Hub: <a href="https://tcga.xenahubs.net" class="uri">https://tcga.xenahubs.net</a>
</li>
<li>GDC Xena Hub: <a href="https://gdc.xenahubs.net" class="uri">https://gdc.xenahubs.net</a>
</li>
<li>ICGC Xena Hub: <a href="https://icgc.xenahubs.net" class="uri">https://icgc.xenahubs.net</a>
</li>
<li>Pan-Cancer Atlas Hub: <a href="https://pancanatlas.xenahubs.net" class="uri">https://pancanatlas.xenahubs.net</a>
</li>
<li>UCSC Toil RNAseq Recompute Compendium Hub: <a href="https://toil.xenahubs.net" class="uri">https://toil.xenahubs.net</a>
</li>
<li>Treehouse Hub: <a href="https://xena.treehouse.gi.ucsc.edu" class="uri">https://xena.treehouse.gi.ucsc.edu</a>
</li>
<li>PCAWG Xena Hub: <a href="https://pcawg.xenahubs.net" class="uri">https://pcawg.xenahubs.net</a>
</li>
<li>ATAC-seq Hub: <a href="https://atacseq.xenahubs.net" class="uri">https://atacseq.xenahubs.net</a>
</li>
<li>Singel Cell Xena Hub: <a href="https://singlecellnew.xenahubs.net" class="uri">https://singlecellnew.xenahubs.net</a>
</li>
<li>Kids First Xena Hub: <a href="https://kidsfirst.xenahubs.net" class="uri">https://kidsfirst.xenahubs.net</a>
</li>
</ul>
<p>Users can update dataset list from the newest version of UCSC Xena by hand with <code><a href="../reference/XenaDataUpdate.html">XenaDataUpdate()</a></code> function, followed by restarting R and <code><a href="https://rdrr.io/r/base/library.html">library(UCSCXenaTools)</a></code>.</p>
<p>If any url of data hub is changed or a new data hub is online, please remind me by emailing to <a href="mailto:w_shixiang@163.com" class="email">w_shixiang@163.com</a> or <a href="https://github.com/ropensci/UCSCXenaTools/issues">opening an issue on GitHub</a>.</p>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<p>Download UCSC Xena datasets and load them into R by <strong>UCSCXenaTools</strong> is a workflow with <code>generate</code>, <code>filter</code>, <code>query</code>, <code>download</code> and <code>prepare</code> 5 steps, which are implemented as <code>XenaGenerate</code>, <code>XenaFilter</code>, <code>XenaQuery</code>, <code>XenaDownload</code> and <code>XenaPrepare</code> functions, respectively. They are very clear and easy to use and combine with other packages like <code>dplyr</code>.</p>
<p>To show the basic usage of <strong>UCSCXenaTools</strong>, we will download clinical data of LUNG, LUAD, LUSC from TCGA (hg19 version) data hub.</p>
<div id="xenadata-data-frame" class="section level3">
<h3 class="hasAnchor">
<a href="#xenadata-data-frame" class="anchor"></a>XenaData data.frame</h3>
<p><strong>UCSCXenaTools</strong> uses a <code>data.frame</code> object (built in package) <code>XenaData</code> to generate an instance of <code>XenaHub</code> class, which records information of all datasets of UCSC Xena Data Hubs.</p>
<p>You can load <code>XenaData</code> after loading <code>UCSCXenaTools</code> into R.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(UCSCXenaTools)</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="co">#&gt; =========================================================================================</span></span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="co">#&gt; UCSCXenaTools version 1.3.0</span></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="co">#&gt; Project URL: https://github.com/ropensci/UCSCXenaTools</span></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="co">#&gt; Usages: https://cran.r-project.org/web/packages/UCSCXenaTools/vignettes/USCSXenaTools.html</span></span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="co">#&gt; </span></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="co">#&gt; If you use it in published research, please cite:</span></span>
<span id="cb4-8"><a href="#cb4-8"></a><span class="co">#&gt; Wang et al., (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data</span></span>
<span id="cb4-9"><a href="#cb4-9"></a><span class="co">#&gt;   from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq.</span></span>
<span id="cb4-10"><a href="#cb4-10"></a><span class="co">#&gt;   Journal of Open Source Software, 4(40), 1627, https://doi.org/10.21105/joss.01627</span></span>
<span id="cb4-11"><a href="#cb4-11"></a><span class="co">#&gt; =========================================================================================</span></span>
<span id="cb4-12"><a href="#cb4-12"></a><span class="co">#&gt;                               --Enjoy it--</span></span>
<span id="cb4-13"><a href="#cb4-13"></a><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(XenaData)</span>
<span id="cb4-14"><a href="#cb4-14"></a></span>
<span id="cb4-15"><a href="#cb4-15"></a><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(XenaData)</span>
<span id="cb4-16"><a href="#cb4-16"></a><span class="co">#&gt; # A tibble: 6 x 17</span></span>
<span id="cb4-17"><a href="#cb4-17"></a><span class="co">#&gt;   XenaHosts XenaHostNames XenaCohorts XenaDatasets SampleCount DataSubtype Label</span></span>
<span id="cb4-18"><a href="#cb4-18"></a><span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;              &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;</span></span>
<span id="cb4-19"><a href="#cb4-19"></a><span class="co">#&gt; 1 https://~ publicHub     Breast Can~ ucsfNeve_pu~          51 gene expre~ Neve~</span></span>
<span id="cb4-20"><a href="#cb4-20"></a><span class="co">#&gt; 2 https://~ publicHub     Breast Can~ ucsfNeve_pu~          57 phenotype   Phen~</span></span>
<span id="cb4-21"><a href="#cb4-21"></a><span class="co">#&gt; 3 https://~ publicHub     Glioma (Ko~ kotliarov20~         194 copy number Kotl~</span></span>
<span id="cb4-22"><a href="#cb4-22"></a><span class="co">#&gt; 4 https://~ publicHub     Glioma (Ko~ kotliarov20~         194 phenotype   Phen~</span></span>
<span id="cb4-23"><a href="#cb4-23"></a><span class="co">#&gt; 5 https://~ publicHub     Lung Cance~ weir2007_pu~         383 copy number CGH  </span></span>
<span id="cb4-24"><a href="#cb4-24"></a><span class="co">#&gt; 6 https://~ publicHub     Lung Cance~ weir2007_pu~         383 phenotype   Phen~</span></span>
<span id="cb4-25"><a href="#cb4-25"></a><span class="co">#&gt; # ... with 10 more variables: Type &lt;chr&gt;, AnatomicalOrigin &lt;chr&gt;,</span></span>
<span id="cb4-26"><a href="#cb4-26"></a><span class="co">#&gt; #   SampleType &lt;chr&gt;, Tags &lt;chr&gt;, ProbeMap &lt;chr&gt;, LongTitle &lt;chr&gt;,</span></span>
<span id="cb4-27"><a href="#cb4-27"></a><span class="co">#&gt; #   Citation &lt;chr&gt;, Version &lt;chr&gt;, Unit &lt;chr&gt;, Platform &lt;chr&gt;</span></span></code></pre></div>
</div>
<div id="workflow" class="section level3">
<h3 class="hasAnchor">
<a href="#workflow" class="anchor"></a>Workflow</h3>
<p>Select datasets.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="co"># The options in XenaFilter function support Regular Expression</span></span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="kw"><a href="../reference/XenaGenerate.html">XenaGenerate</a></span>(<span class="dt">subset =</span> XenaHostNames<span class="op">==</span><span class="st">"tcgaHub"</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="st">  </span><span class="kw"><a href="../reference/XenaFilter.html">XenaFilter</a></span>(<span class="dt">filterDatasets =</span> <span class="st">"clinical"</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="st">  </span><span class="kw"><a href="../reference/XenaFilter.html">XenaFilter</a></span>(<span class="dt">filterDatasets =</span> <span class="st">"LUAD|LUSC|LUNG"</span>) -&gt;<span class="st"> </span>df_todo</span>
<span id="cb5-5"><a href="#cb5-5"></a></span>
<span id="cb5-6"><a href="#cb5-6"></a>df_todo</span>
<span id="cb5-7"><a href="#cb5-7"></a><span class="co">#&gt; class: XenaHub </span></span>
<span id="cb5-8"><a href="#cb5-8"></a><span class="co">#&gt; hosts():</span></span>
<span id="cb5-9"><a href="#cb5-9"></a><span class="co">#&gt;   https://tcga.xenahubs.net</span></span>
<span id="cb5-10"><a href="#cb5-10"></a><span class="co">#&gt; cohorts() (3 total):</span></span>
<span id="cb5-11"><a href="#cb5-11"></a><span class="co">#&gt;   TCGA Lung Cancer (LUNG)</span></span>
<span id="cb5-12"><a href="#cb5-12"></a><span class="co">#&gt;   TCGA Lung Adenocarcinoma (LUAD)</span></span>
<span id="cb5-13"><a href="#cb5-13"></a><span class="co">#&gt;   TCGA Lung Squamous Cell Carcinoma (LUSC)</span></span>
<span id="cb5-14"><a href="#cb5-14"></a><span class="co">#&gt; datasets() (3 total):</span></span>
<span id="cb5-15"><a href="#cb5-15"></a><span class="co">#&gt;   TCGA.LUNG.sampleMap/LUNG_clinicalMatrix</span></span>
<span id="cb5-16"><a href="#cb5-16"></a><span class="co">#&gt;   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</span></span>
<span id="cb5-17"><a href="#cb5-17"></a><span class="co">#&gt;   TCGA.LUSC.sampleMap/LUSC_clinicalMatrix</span></span></code></pre></div>
<p>Sometimes we only know some keywords, <code><a href="../reference/XenaScan.html">XenaScan()</a></code> can be used to scan all rows to detect if the keywords exist in <code>XenaData</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>x1 =<span class="st"> </span><span class="kw"><a href="../reference/XenaScan.html">XenaScan</a></span>(<span class="dt">pattern =</span> <span class="st">'Blood'</span>)</span>
<span id="cb6-2"><a href="#cb6-2"></a>x2 =<span class="st"> </span><span class="kw"><a href="../reference/XenaScan.html">XenaScan</a></span>(<span class="dt">pattern =</span> <span class="st">'LUNG'</span>, <span class="dt">ignore.case =</span> <span class="ot">FALSE</span>)</span>
<span id="cb6-3"><a href="#cb6-3"></a></span>
<span id="cb6-4"><a href="#cb6-4"></a>x1 <span class="op">%&gt;%</span></span>
<span id="cb6-5"><a href="#cb6-5"></a><span class="st">    </span><span class="kw"><a href="../reference/XenaGenerate.html">XenaGenerate</a></span>()</span>
<span id="cb6-6"><a href="#cb6-6"></a><span class="co">#&gt; class: XenaHub </span></span>
<span id="cb6-7"><a href="#cb6-7"></a><span class="co">#&gt; hosts():</span></span>
<span id="cb6-8"><a href="#cb6-8"></a><span class="co">#&gt;   https://ucscpublic.xenahubs.net</span></span>
<span id="cb6-9"><a href="#cb6-9"></a><span class="co">#&gt;   https://tcga.xenahubs.net</span></span>
<span id="cb6-10"><a href="#cb6-10"></a><span class="co">#&gt; cohorts() (6 total):</span></span>
<span id="cb6-11"><a href="#cb6-11"></a><span class="co">#&gt;   Connectivity Map</span></span>
<span id="cb6-12"><a href="#cb6-12"></a><span class="co">#&gt;   TARGET Acute Lymphoblastic Leukemia</span></span>
<span id="cb6-13"><a href="#cb6-13"></a><span class="co">#&gt;   Pediatric tumor (Khan)</span></span>
<span id="cb6-14"><a href="#cb6-14"></a><span class="co">#&gt;   Acute lymphoblastic leukemia (Mullighan 2008)</span></span>
<span id="cb6-15"><a href="#cb6-15"></a><span class="co">#&gt;   TCGA Pan-Cancer (PANCAN)</span></span>
<span id="cb6-16"><a href="#cb6-16"></a><span class="co">#&gt;   TCGA Acute Myeloid Leukemia (LAML)</span></span>
<span id="cb6-17"><a href="#cb6-17"></a><span class="co">#&gt; datasets() (28 total):</span></span>
<span id="cb6-18"><a href="#cb6-18"></a><span class="co">#&gt;   cmap/rankMatrix_reverse</span></span>
<span id="cb6-19"><a href="#cb6-19"></a><span class="co">#&gt;   TARGET_ALL/TARGETcnv_genomicMatrix</span></span>
<span id="cb6-20"><a href="#cb6-20"></a><span class="co">#&gt;   TARGET_ALL/TARGETexp_genomicMatrix</span></span>
<span id="cb6-21"><a href="#cb6-21"></a><span class="co">#&gt;   ...</span></span>
<span id="cb6-22"><a href="#cb6-22"></a><span class="co">#&gt;   TCGA.LAML.sampleMap/mutation_wustl</span></span>
<span id="cb6-23"><a href="#cb6-23"></a><span class="co">#&gt;   TCGA.LAML.sampleMap/mutation_wustl_gene</span></span>
<span id="cb6-24"><a href="#cb6-24"></a>x2 <span class="op">%&gt;%</span></span>
<span id="cb6-25"><a href="#cb6-25"></a><span class="st">    </span><span class="kw"><a href="../reference/XenaGenerate.html">XenaGenerate</a></span>()</span>
<span id="cb6-26"><a href="#cb6-26"></a><span class="co">#&gt; class: XenaHub </span></span>
<span id="cb6-27"><a href="#cb6-27"></a><span class="co">#&gt; hosts():</span></span>
<span id="cb6-28"><a href="#cb6-28"></a><span class="co">#&gt;   https://tcga.xenahubs.net</span></span>
<span id="cb6-29"><a href="#cb6-29"></a><span class="co">#&gt; cohorts() (1 total):</span></span>
<span id="cb6-30"><a href="#cb6-30"></a><span class="co">#&gt;   TCGA Lung Cancer (LUNG)</span></span>
<span id="cb6-31"><a href="#cb6-31"></a><span class="co">#&gt; datasets() (15 total):</span></span>
<span id="cb6-32"><a href="#cb6-32"></a><span class="co">#&gt;   TCGA.LUNG.sampleMap/HumanMethylation27</span></span>
<span id="cb6-33"><a href="#cb6-33"></a><span class="co">#&gt;   TCGA.LUNG.sampleMap/HumanMethylation450</span></span>
<span id="cb6-34"><a href="#cb6-34"></a><span class="co">#&gt;   TCGA.LUNG.sampleMap/Gistic2_CopyNumber_Gistic2_all_data_by_genes</span></span>
<span id="cb6-35"><a href="#cb6-35"></a><span class="co">#&gt;   ...</span></span>
<span id="cb6-36"><a href="#cb6-36"></a><span class="co">#&gt;   mc3/LUNG_mc3.txt</span></span>
<span id="cb6-37"><a href="#cb6-37"></a><span class="co">#&gt;   mc3_gene_level/LUNG_mc3_gene_level.txt</span></span></code></pre></div>
<p>Query and download.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="kw"><a href="../reference/XenaQuery.html">XenaQuery</a></span>(df_todo) <span class="op">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="st">  </span><span class="kw"><a href="../reference/XenaDownload.html">XenaDownload</a></span>() -&gt;<span class="st"> </span>xe_download</span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="co">#&gt; This will check url status, please be patient.</span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co">#&gt; All downloaded files will under directory D:/Tool/Rtmp\RtmpamNIp2.</span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="co">#&gt; The 'trans_slash' option is FALSE, keep same directory structure as Xena.</span></span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="co">#&gt; Creating directories for datasets...</span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="co">#&gt; Downloading TCGA.LUNG.sampleMap/LUNG_clinicalMatrix</span></span>
<span id="cb7-8"><a href="#cb7-8"></a><span class="co">#&gt; Downloading TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</span></span>
<span id="cb7-9"><a href="#cb7-9"></a><span class="co">#&gt; Downloading TCGA.LUSC.sampleMap/LUSC_clinicalMatrix</span></span></code></pre></div>
<p>Prepare data into R for analysis.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>cli =<span class="st"> </span><span class="kw"><a href="../reference/XenaPrepare.html">XenaPrepare</a></span>(xe_download)</span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="kw"><a href="https://rdrr.io/r/base/class.html">class</a></span>(cli)</span>
<span id="cb8-3"><a href="#cb8-3"></a><span class="co">#&gt; [1] "list"</span></span>
<span id="cb8-4"><a href="#cb8-4"></a><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(cli)</span>
<span id="cb8-5"><a href="#cb8-5"></a><span class="co">#&gt; [1] "LUNG_clinicalMatrix" "LUAD_clinicalMatrix" "LUSC_clinicalMatrix"</span></span></code></pre></div>
</div>
<div id="browse-datasets" class="section level3">
<h3 class="hasAnchor">
<a href="#browse-datasets" class="anchor"></a>Browse datasets</h3>
<p>Create two XenaHub objects:</p>
<ul>
<li>
<code>to_browse</code> - a XenaHub object containing a cohort and a dataset.</li>
<li>
<code>to_browse2</code> - a XenaHub object containing 2 cohorts and 2 datasets.</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="kw"><a href="../reference/XenaGenerate.html">XenaGenerate</a></span>(<span class="dt">subset =</span> XenaHostNames<span class="op">==</span><span class="st">"tcgaHub"</span>) <span class="op">%&gt;%</span></span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="st">    </span><span class="kw"><a href="../reference/XenaFilter.html">XenaFilter</a></span>(<span class="dt">filterDatasets =</span> <span class="st">"clinical"</span>) <span class="op">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3"></a><span class="st">    </span><span class="kw"><a href="../reference/XenaFilter.html">XenaFilter</a></span>(<span class="dt">filterDatasets =</span> <span class="st">"LUAD"</span>) -&gt;<span class="st"> </span>to_browse</span>
<span id="cb9-4"><a href="#cb9-4"></a></span>
<span id="cb9-5"><a href="#cb9-5"></a>to_browse</span>
<span id="cb9-6"><a href="#cb9-6"></a><span class="co">#&gt; class: XenaHub </span></span>
<span id="cb9-7"><a href="#cb9-7"></a><span class="co">#&gt; hosts():</span></span>
<span id="cb9-8"><a href="#cb9-8"></a><span class="co">#&gt;   https://tcga.xenahubs.net</span></span>
<span id="cb9-9"><a href="#cb9-9"></a><span class="co">#&gt; cohorts() (1 total):</span></span>
<span id="cb9-10"><a href="#cb9-10"></a><span class="co">#&gt;   TCGA Lung Adenocarcinoma (LUAD)</span></span>
<span id="cb9-11"><a href="#cb9-11"></a><span class="co">#&gt; datasets() (1 total):</span></span>
<span id="cb9-12"><a href="#cb9-12"></a><span class="co">#&gt;   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</span></span>
<span id="cb9-13"><a href="#cb9-13"></a></span>
<span id="cb9-14"><a href="#cb9-14"></a><span class="kw"><a href="../reference/XenaGenerate.html">XenaGenerate</a></span>(<span class="dt">subset =</span> XenaHostNames<span class="op">==</span><span class="st">"tcgaHub"</span>) <span class="op">%&gt;%</span></span>
<span id="cb9-15"><a href="#cb9-15"></a><span class="st">    </span><span class="kw"><a href="../reference/XenaFilter.html">XenaFilter</a></span>(<span class="dt">filterDatasets =</span> <span class="st">"clinical"</span>) <span class="op">%&gt;%</span></span>
<span id="cb9-16"><a href="#cb9-16"></a><span class="st">    </span><span class="kw"><a href="../reference/XenaFilter.html">XenaFilter</a></span>(<span class="dt">filterDatasets =</span> <span class="st">"LUAD|LUSC"</span>) -&gt;<span class="st"> </span>to_browse2</span>
<span id="cb9-17"><a href="#cb9-17"></a></span>
<span id="cb9-18"><a href="#cb9-18"></a>to_browse2</span>
<span id="cb9-19"><a href="#cb9-19"></a><span class="co">#&gt; class: XenaHub </span></span>
<span id="cb9-20"><a href="#cb9-20"></a><span class="co">#&gt; hosts():</span></span>
<span id="cb9-21"><a href="#cb9-21"></a><span class="co">#&gt;   https://tcga.xenahubs.net</span></span>
<span id="cb9-22"><a href="#cb9-22"></a><span class="co">#&gt; cohorts() (2 total):</span></span>
<span id="cb9-23"><a href="#cb9-23"></a><span class="co">#&gt;   TCGA Lung Adenocarcinoma (LUAD)</span></span>
<span id="cb9-24"><a href="#cb9-24"></a><span class="co">#&gt;   TCGA Lung Squamous Cell Carcinoma (LUSC)</span></span>
<span id="cb9-25"><a href="#cb9-25"></a><span class="co">#&gt; datasets() (2 total):</span></span>
<span id="cb9-26"><a href="#cb9-26"></a><span class="co">#&gt;   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</span></span>
<span id="cb9-27"><a href="#cb9-27"></a><span class="co">#&gt;   TCGA.LUSC.sampleMap/LUSC_clinicalMatrix</span></span></code></pre></div>
<p><code><a href="../reference/XenaBrowse.html">XenaBrowse()</a></code> function can be used to browse dataset/cohort links using your default web browser. At default, this function limits one dataset/cohort for preventing user to open too many links at once.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="co"># This will open you web browser</span></span>
<span id="cb10-2"><a href="#cb10-2"></a><span class="kw"><a href="../reference/XenaBrowse.html">XenaBrowse</a></span>(to_browse)</span>
<span id="cb10-3"><a href="#cb10-3"></a></span>
<span id="cb10-4"><a href="#cb10-4"></a><span class="kw"><a href="../reference/XenaBrowse.html">XenaBrowse</a></span>(to_browse, <span class="dt">type =</span> <span class="st">"cohort"</span>)</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="co"># This will throw error</span></span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="kw"><a href="../reference/XenaBrowse.html">XenaBrowse</a></span>(to_browse2)</span>
<span id="cb11-3"><a href="#cb11-3"></a><span class="co">#&gt; Error in XenaBrowse(to_browse2): This function limite 1 dataset to browse.</span></span>
<span id="cb11-4"><a href="#cb11-4"></a><span class="co">#&gt;  Set multiple to TRUE if you want to browse multiple links.</span></span>
<span id="cb11-5"><a href="#cb11-5"></a></span>
<span id="cb11-6"><a href="#cb11-6"></a><span class="kw"><a href="../reference/XenaBrowse.html">XenaBrowse</a></span>(to_browse2, <span class="dt">type =</span> <span class="st">"cohort"</span>)</span>
<span id="cb11-7"><a href="#cb11-7"></a><span class="co">#&gt; Error in XenaBrowse(to_browse2, type = "cohort"): This function limite 1 cohort to browse. </span></span>
<span id="cb11-8"><a href="#cb11-8"></a><span class="co">#&gt;  Set multiple to TRUE if you want to browse multiple links.</span></span></code></pre></div>
<p>When you make sure you want to open multiple links, you can set <code>multiple</code> option to <code>TRUE</code>.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="kw"><a href="../reference/XenaBrowse.html">XenaBrowse</a></span>(to_browse2, <span class="dt">multiple =</span> <span class="ot">TRUE</span>)</span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="kw"><a href="../reference/XenaBrowse.html">XenaBrowse</a></span>(to_browse2, <span class="dt">type =</span> <span class="st">"cohort"</span>, <span class="dt">multiple =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
</div>
</div>
<div id="more-usages" class="section level2">
<h2 class="hasAnchor">
<a href="#more-usages" class="anchor"></a>More usages</h2>
<p>The core functionality has been described above. I write more usages about this package in my website but not here because sometimes package check will fail due to internet problem.</p>
<ul>
<li>
<a href="https://shixiangwang.github.io/home/en/tools/ucscxenatools-intro/">Introduction and basic usage of UCSCXenaTools</a> - <a href="https://shixiangwang.github.io/home/en/tools/ucscxenatools-intro.pdf">PDF</a>
</li>
<li>
<a href="https://shixiangwang.github.io/home/en/tools/ucscxenatools-api/">APIs of UCSCXenaTools</a> - <a href="https://shixiangwang.github.io/home/en/tools/ucscxenatools-api.pdf">PDF</a>
</li>
</ul>
<p>A use case for survival analysis has been published on rOpenSci, please read <a href="https://ropensci.org/technotes/2019/09/06/ucscxenatools-surv/">UCSCXenaTools: Retrieve Gene Expression and Clinical Information from UCSC Xena for Survival Analysis</a>.</p>
</div>
<div id="qa" class="section level2">
<h2 class="hasAnchor">
<a href="#qa" class="anchor"></a>QA</h2>
<div id="how-to-resume-file-from-breakpoint" class="section level3">
<h3 class="hasAnchor">
<a href="#how-to-resume-file-from-breakpoint" class="anchor"></a>How to resume file from breakpoint</h3>
<p>Thanks to the UCSC Xena team, the new feature ‘resume from breakpoint’ is added and can be done by <strong>XenaDownload()</strong> with the <code>method</code> and <code>extra</code> flags specified.</p>
<p>Of note, the corresponding <code>wget</code> or <code>curl</code> command must be installed by your OS and can be found by R.</p>
<p>The folliwng code gives a test example, the data can be viewed on <a href="https://xenabrowser.net/datapages/?dataset=TcgaTargetGtex_expected_count&amp;host=https%3A%2F%2Ftoil.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">web page</a>.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(UCSCXenaTools)</span>
<span id="cb13-2"><a href="#cb13-2"></a>xe =<span class="st"> </span><span class="kw"><a href="../reference/XenaGenerate.html">XenaGenerate</a></span>(<span class="dt">subset =</span> XenaDatasets <span class="op">==</span><span class="st"> "TcgaTargetGtex_expected_count"</span>)</span>
<span id="cb13-3"><a href="#cb13-3"></a>xe</span>
<span id="cb13-4"><a href="#cb13-4"></a>xq =<span class="st"> </span><span class="kw"><a href="../reference/XenaQuery.html">XenaQuery</a></span>(xe)</span>
<span id="cb13-5"><a href="#cb13-5"></a><span class="co"># You cannot resume from breakpoint in default mode</span></span>
<span id="cb13-6"><a href="#cb13-6"></a><span class="kw"><a href="../reference/XenaDownload.html">XenaDownload</a></span>(xq, <span class="dt">destdir =</span> <span class="st">"~/test/"</span>, <span class="dt">force =</span> <span class="ot">TRUE</span>)</span>
<span id="cb13-7"><a href="#cb13-7"></a><span class="co"># You can do it with 'curl' command</span></span>
<span id="cb13-8"><a href="#cb13-8"></a><span class="kw"><a href="../reference/XenaDownload.html">XenaDownload</a></span>(xq, <span class="dt">destdir =</span> <span class="st">"~/test/"</span>, <span class="dt">method =</span> <span class="st">"curl"</span>, <span class="dt">extra =</span> <span class="st">"-C -"</span>, <span class="dt">force =</span> <span class="ot">TRUE</span>)</span>
<span id="cb13-9"><a href="#cb13-9"></a><span class="co"># You can do it with 'wget' command</span></span>
<span id="cb13-10"><a href="#cb13-10"></a><span class="kw"><a href="../reference/XenaDownload.html">XenaDownload</a></span>(xq, <span class="dt">destdir =</span> <span class="st">"~/test/"</span>, <span class="dt">method =</span> <span class="st">"wget"</span>, <span class="dt">extra =</span> <span class="st">"-c"</span>, <span class="dt">force =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
</div>
</div>
<div id="citation" class="section level2">
<h2 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h2>
<p>Cite me by the following paper.</p>
<pre><code>Wang et al., (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data
  from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq. 
  Journal of Open Source Software, 4(40), 1627, https://doi.org/10.21105/joss.01627

# For BibTex
  
@article{Wang2019UCSCXenaTools,
    journal = {Journal of Open Source Software},
    doi = {10.21105/joss.01627},
    issn = {2475-9066},
    number = {40},
    publisher = {The Open Journal},
    title = {The UCSCXenaTools R package: a toolkit for accessing genomics data from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq},
    url = {http://dx.doi.org/10.21105/joss.01627},
    volume = {4},
    author = {Wang, Shixiang and Liu, Xuesong},
    pages = {1627},
    date = {2019-08-05},
    year = {2019},
    month = {8},
    day = {5},
}</code></pre>
<p>Cite UCSC Xena by the following paper.</p>
<pre><code>Goldman, Mary, et al. "The UCSC Xena Platform for cancer genomics data 
    visualization and interpretation." BioRxiv (2019): 326470.</code></pre>
</div>
<div id="acknowledgments" class="section level2">
<h2 class="hasAnchor">
<a href="#acknowledgments" class="anchor"></a>Acknowledgments</h2>
<p>This package is based on <a href="https://github.com/mtmorgan/XenaR">XenaR</a>, thanks <a href="https://github.com/mtmorgan">Martin Morgan</a> for his work.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#installation">Installation</a></li>
      <li><a href="#data-hub-list">Data Hub List</a></li>
      <li><a href="#usage">Usage</a></li>
      <li><a href="#more-usages">More usages</a></li>
      <li><a href="#qa">QA</a></li>
      <li><a href="#citation">Citation</a></li>
      <li><a href="#acknowledgments">Acknowledgments</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://shixiangwang.github.io/">Shixiang Wang</a>, Xue-Song Liu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
