<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Fetch Data from UCSC Xena Hosts — fetch • UCSCXenaTools</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Fetch Data from UCSC Xena Hosts — fetch"><meta property="og:description" content="When you want to query just data for several genes/samples from UCSC Xena datasets, a better way
is to use these fetch_ functions instead of downloading a whole dataset. Details about functions
please see the following sections."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">UCSCXenaTools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.4.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/USCSXenaTools.html">UCSCXenaTools: an R package for Accessing Genomics Data from UCSC Xena platform, from Cancer Multi-omics to Single-cell RNA-seq</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ropensci/UCSCXenaTools/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Fetch Data from UCSC Xena Hosts</h1>
    <small class="dont-index">Source: <a href="https://github.com/ropensci/UCSCXenaTools/blob/HEAD/R/fetch.R" class="external-link"><code>R/fetch.R</code></a></small>
    <div class="hidden name"><code>fetch.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>When you want to query just data for several genes/samples from UCSC Xena datasets, a better way
is to use these <code>fetch_</code> functions instead of downloading a whole dataset. Details about functions
please see the following sections.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">fetch</span><span class="op">(</span><span class="va">host</span>, <span class="va">dataset</span><span class="op">)</span>

<span class="fu">fetch_dense_values</span><span class="op">(</span>
  <span class="va">host</span>,
  <span class="va">dataset</span>,
  identifiers <span class="op">=</span> <span class="cn">NULL</span>,
  samples <span class="op">=</span> <span class="cn">NULL</span>,
  check <span class="op">=</span> <span class="cn">TRUE</span>,
  use_probeMap <span class="op">=</span> <span class="cn">FALSE</span>,
  time_limit <span class="op">=</span> <span class="fl">30</span>
<span class="op">)</span>

<span class="fu">fetch_sparse_values</span><span class="op">(</span><span class="va">host</span>, <span class="va">dataset</span>, <span class="va">genes</span>, samples <span class="op">=</span> <span class="cn">NULL</span>, time_limit <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>

<span class="fu">fetch_dataset_samples</span><span class="op">(</span><span class="va">host</span>, <span class="va">dataset</span>, limit <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>

<span class="fu">fetch_dataset_identifiers</span><span class="op">(</span><span class="va">host</span>, <span class="va">dataset</span><span class="op">)</span>

<span class="fu">has_probeMap</span><span class="op">(</span><span class="va">host</span>, <span class="va">dataset</span>, return_url <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>host</dt>
<dd><p>a UCSC Xena host, like "https://toil.xenahubs.net".
All available hosts can be printed by <code><a href="xena_default_hosts.html">xena_default_hosts()</a></code>.</p></dd>


<dt>dataset</dt>
<dd><p>a UCSC Xena dataset, like "tcga_RSEM_gene_tpm".
All available datasets can be printed by running <code>XenaData$XenaDatasets</code> or
obtained from <a href="https://xenabrowser.net/datapages/" class="external-link">UCSC Xena datapages</a>.</p></dd>


<dt>identifiers</dt>
<dd><p>Identifiers could be probe (like "ENSG00000000419.12"),
gene (like "TP53") etc.. If it is <code>NULL</code>, all identifiers in the dataset will be used.</p></dd>


<dt>samples</dt>
<dd><p>ID of samples, like "TCGA-02-0047-01".
If it is <code>NULL</code>, all samples in the dataset will be used. However, it is better to download
the whole datasets if you query many samples and genes.</p></dd>


<dt>check</dt>
<dd><p>if <code>TRUE</code>, check whether specified <code>identifiers</code> and <code>samples</code> exist the dataset
(all failed items will be filtered out). However, if <code>FALSE</code>, the code is much faster.</p></dd>


<dt>use_probeMap</dt>
<dd><p>if <code>TRUE</code>, will check if the dataset has ProbeMap firstly.
When the dataset you want to query has a identifier-to-gene mapping, identifiers can be
gene symbols even the identifiers of dataset are probes or others.</p></dd>


<dt>time_limit</dt>
<dd><p>time limit for getting response in seconds.</p></dd>


<dt>genes</dt>
<dd><p>gene names.</p></dd>


<dt>limit</dt>
<dd><p>number of samples, if <code>NULL</code>, return all samples.</p></dd>


<dt>return_url</dt>
<dd><p>if <code>TRUE</code>, returns the info of probeMap
instead of a logical value when the result exists.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a <code>matirx</code> or character vector or a <code>list</code>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>There are three primary data types: dense matrix (samples by probes (or say identifiers)),
sparse (sample, position, variant), and segmented (sample, position, value).</p>
<p>Dense matrices can be genotypic or phenotypic, it is a sample-by-identifiers matrix.
Phenotypic matrices have associated field metadata (descriptive names, codes, etc.).
Genotypic matricies may have an associated probeMap, which maps probes to genomic locations.
If a matrix has hugo probeMap, the probes themselves are gene names. Otherwise, a probeMap is
used to map a gene location to a set of probes.</p>
    </div>
    <div id="functions">
    <h2>Functions</h2>
    
<ul><li><p><code>fetch_dense_values</code>: fetches values from a dense matrix.</p></li>
<li><p><code>fetch_sparse_values</code>: fetches values from a sparse <code>data.frame</code>.</p></li>
<li><p><code>fetch_dataset_samples</code>: fetches samples from a dataset</p></li>
<li><p><code>fetch_dataset_identifiers</code>: fetches identifies from a dataset.</p></li>
<li><p><code>has_probeMap</code>: checks if a dataset has ProbeMap.</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/UCSCXenaTools/" class="external-link">UCSCXenaTools</a></span><span class="op">)</span>

<span class="va">host</span> <span class="op">&lt;-</span> <span class="st">"https://toil.xenahubs.net"</span>
<span class="va">dataset</span> <span class="op">&lt;-</span> <span class="st">"tcga_RSEM_gene_tpm"</span>
<span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TCGA-02-0047-01"</span>, <span class="st">"TCGA-02-0055-01"</span>, <span class="st">"TCGA-02-2483-01"</span>, <span class="st">"TCGA-02-2485-01"</span><span class="op">)</span>
<span class="va">probes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSG00000282740.1"</span>, <span class="st">"ENSG00000000005.5"</span>, <span class="st">"ENSG00000000419.12"</span><span class="op">)</span>
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TP53"</span>, <span class="st">"RB1"</span>, <span class="st">"PIK3CA"</span><span class="op">)</span>

<span class="co"># \donttest{</span>
<span class="co"># Fetch samples</span>
<span class="fu">fetch_dataset_samples</span><span class="op">(</span><span class="va">host</span>, <span class="va">dataset</span>, <span class="fl">2</span><span class="op">)</span>
<span class="co"># Fetch identifiers</span>
<span class="fu">fetch_dataset_identifiers</span><span class="op">(</span><span class="va">host</span>, <span class="va">dataset</span><span class="op">)</span>
<span class="co"># Fetch expression value by probes</span>
<span class="fu">fetch_dense_values</span><span class="op">(</span><span class="va">host</span>, <span class="va">dataset</span>, <span class="va">probes</span>, <span class="va">samples</span>, check <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co"># Fetch expression value by gene symbol (if the dataset has probeMap)</span>
<span class="fu">has_probeMap</span><span class="op">(</span><span class="va">host</span>, <span class="va">dataset</span><span class="op">)</span>
<span class="fu">fetch_dense_values</span><span class="op">(</span><span class="va">host</span>, <span class="va">dataset</span>, <span class="va">genes</span>, <span class="va">samples</span>, check <span class="op">=</span> <span class="cn">FALSE</span>, use_probeMap <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co"># }</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://shixiangwang.github.io/" class="external-link">Shixiang Wang</a>, Xue-Song Liu.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.</p>
</div>

      </footer></div>

  


  

  </body></html>

